pkg/config/process_test.go

```diff
@@ -0,0 +1,173 @@
+
+	for envVar := range envVars {
+		if !strings.HasPrefix(envVar, "DD_PROCESS_AGENT") {
+			continue
+		}
+
+		processAgentEnvVar := strings.Replace(envVar, "PROCESS_AGENT", "PROCESS_CONFIG", 1)
+		t.Run(fmt.Sprintf("%s and %s", envVar, processAgentEnvVar), func(t *testing.T) {
+			// Check to see if envVars contains processAgentEnvVar. We can't use assert.Contains,
+			// because when it fails the library prints all of envVars which is too noisy
+			_, ok := envVars[processAgentEnvVar]
+			assert.Truef(t, ok, "%s is defined but not %s", envVar, processAgentEnvVar)
+		})
+	}
+}
+
+func TestEnvVarOverride(t *testing.T) {
+	cfg := setupConf()
+
+	reset := setEnvForTest("DD_LOG_LEVEL", "warn")
+	assert.Equal(t, "warn", cfg.GetString("log_level"))
+	reset()
+
+	reset = setEnvForTest("DD_LOG_TO_CONSOLE", "false")
+	assert.False(t, cfg.GetBool("log_to_console"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_CONFIG_LOG_FILE", "test")
+	assert.Equal(t, "test", cfg.GetString("process_config.log_file"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_AGENT_LOG_FILE", "test1")
+	assert.Equal(t, "test1", cfg.GetString("process_config.log_file"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_CONFIG_DD_AGENT_BIN", "test")
+	assert.Equal(t, "test", cfg.GetString("process_config.dd_agent_bin"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_AGENT_DD_AGENT_BIN", "test1")
+	assert.Equal(t, "test1", cfg.GetString("process_config.dd_agent_bin"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_CONFIG_REMOTE_TAGGER", "false")
+	assert.False(t, cfg.GetBool("process_config.remote_tagger"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_AGENT_REMOTE_TAGGER", "false")
+	assert.False(t, cfg.GetBool("process_config.remote_tagger"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_CONFIG_GRPC_CONNECTION_TIMEOUT_SECS", "1")
+	assert.Equal(t, 1, cfg.GetInt("process_config.grpc_connection_timeout_secs"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_AGENT_GRPC_CONNECTION_TIMEOUT_SECS", "1")
+	assert.Equal(t, 1, cfg.GetInt("process_config.grpc_connection_timeout_secs"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_CONFIG_PROCESS_DISCOVERY_ENABLED", "true")
+	assert.True(t, cfg.GetBool("process_config.process_discovery.enabled"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_AGENT_PROCESS_DISCOVERY_ENABLED", "true")
+	assert.True(t, cfg.GetBool("process_config.process_discovery.enabled"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_CONFIG_PROCESS_DISCOVERY_INTERVAL", "1h")
+	assert.Equal(t, time.Hour, cfg.GetDuration("process_config.process_discovery.interval"))
+	reset()
+
+	reset = setEnvForTest("DD_PROCESS_AGENT_PROCESS_DISCOVERY_INTERVAL", "1h")
+	assert.Equal(t, time.Hour, cfg.GetDuration("process_config.process_discovery.interval"))
+	reset()
+}
+
+func TestProcBindEnvAndSetDefault(t *testing.T) {
+	cfg := setupConf()
+	procBindEnvAndSetDefault(cfg, "process_config.foo.bar", "asdf", "BAZ")
+	assert.Contains(t, cfg.GetEnvVars(), "DD_PROCESS_CONFIG_FOO_BAR")
```

Comment:
No issue with `Contains` here?

Reply:
Yeah the issue is still present. I will rewrite this to use a map like in`TestPrefixes`